src/main/java/courseManager/Outline.java

```diff
@@ -0,0 +1,98 @@
+package courseManager;
+
+import weightScheme.WeightScheme;
+
+import java.util.ArrayList;
+
+public class Outline {
+    private double runningGrade;
+    private double hypothethicalGrade;
+
+    private double percentageCompleted;
+    private int numberOfAssessmentsCompleted; /*do these need to be their own instance variables?*/
+    private ArrayList<Assessment> assessments;
+
+    public Outline () {
+        this.runningGrade = 0.0f;
+        this.hypothethicalGrade = 0.0f;
+        this.assessments = new ArrayList<>();
+    }
+
+    public void setRunningGrade(double runningGrade) {
+
+        this.runningGrade = runningGrade;
+    }
+
+    public void setHypothethicalGrade(double hypothethicalGrade) {
+
+        this.hypothethicalGrade = hypothethicalGrade;
+    }
+
+    public void addAssessment(Assessment assessment) {
+
+        assessments.add(assessment);
+    }
+
+    public void removeAssessment(Assessment assessment) {
+
+        assessments.remove(assessment);
+    }
+
+    public void editAssessment(int index, String title) {
```

Comment:
Similar to my comment in InstanceList, you can just straight up return the assessment itself, it's already decently encapsulated and the only thing accessing it should be the interactor.

Reply:
you're right, i now removed the editAssessment methods. I think we could just do it through the getAssessment method to access the Assessment and then use the set methods